Update vagrant & travis provision.

odolbeau opened the pull request:
Support of autoreconf have been removed from rabbitmq-c. Use cmake instead.

odolbeau commented:
Works great for the vagrant vm.

Same provision script for travis works great on swarrot (see: https://github.com/swarrot/swarrot/blob/master/tests/bin/install_rabbitmq-c.sh)

Looks like it's not working here due to the `sudo: false` in the `.travis.yml` I think.
Do you see any way to avoid this problem?

lstrojny commented on provision/install_rabbitmq-c.sh:
```diff
@@ -1,24 +1,33 @@
+# * https://github.com/alanxz/rabbitmq-c/blob/master/.travis.yml
 
+echo Installing clang ...
+wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
```
Let’s use https here

pinepain replied:
Could you clarify a bit what `llvm` is used here for?

lstrojny commented on provision/install_rabbitmq-c.sh:
```diff
@@ -1,24 +1,33 @@
 
+echo Installing clang ...
+wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+apt-add-repository "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-3.9 main"
```
… and here

lstrojny commented on provision/install_rabbitmq-c.sh:
```diff
@@ -1,24 +1,33 @@
+echo Installing clang ...
+wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+apt-add-repository "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-3.9 main"
+apt-get -q update;
```
Superfluous semicolon

lstrojny commented on provision/install_rabbitmq-c.sh:
```diff
@@ -1,24 +1,33 @@
+wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+apt-add-repository "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-3.9 main"
+apt-get -q update;
+apt-get install -y clang-3.9 libpopt-dev;
```
Superfluous semicolon

pinepain replied:
And what is `clang` supposed to be used?

lstrojny commented on provision/install_rabbitmq-c.sh:
```diff
@@ -1,24 +1,33 @@
+apt-get install -y clang-3.9 libpopt-dev;
 
-LIBRABBITMQ_VERSION=$1
+echo Installing rabbitmq-c ...
```
Please quote the message

pinepain commented on .travis.yml:
```diff
@@ -54,7 +54,7 @@ matrix:
 
 before_install:
-  - sh provision/install_rabbitmq-c.sh ${LIBRABBITMQ_VERSION}
+  - sh -c 'provision/install_rabbitmq-c.sh'
```
Is it intentional that `LIBRABBITMQ_VERSION` removed from args?

odolbeau replied:
The argument wasn't used in the script which directly use the environment variable.

odolbeau replied:
My bad, it was using this argument. I can add it back if you prefer.

odolbeau commented:
I copy / pasted the `before_install` part from rabbitmq-c ([here](https://github.com/alanxz/rabbitmq-c/blob/master/.travis.yml#L51-L56)).
Copy / pasting without thinking is bad! :/

`clang` could be used instead of `gcc` according to the README. `gcc` is fine, I removed this part.
`libopt` is a **optional** dependency => I removed it.

pinepain commented on .travis.yml:
```diff
@@ -54,7 +54,7 @@ matrix:
 
 before_install:
-  - sh provision/install_rabbitmq-c.sh ${LIBRABBITMQ_VERSION}
+  - sh -c './provision/install_rabbitmq-c.sh'
```
I still think that `${LIBRABBITMQ_VERSION}` was removed by accident. Or did I miss something? 

The initial idea was that we pass desired version to install script and later read it from `$1`.

pinepain commented on .travis.yml:
```diff
@@ -54,7 +54,7 @@ matrix:
 
 before_install:
-  - sh provision/install_rabbitmq-c.sh ${LIBRABBITMQ_VERSION}
+  - sudo ./provision/install_rabbitmq-c.sh
```
Ughh, commenting on this for the third time, I still can't get used to new GH review features or my comments get lost somehow. 

It looks like `LIBRABBITMQ_VERSION` was removed by accident as it value later retrieved in install script as `$1`.

Also, I'm not entirely sure that sudo is required to run this script. The install script use `sudo` within for commands that need it.

odolbeau replied:
I removed the argument to use directly the `LIBRABBITMQ_VERSION` environment variable inside the script.
I can add the argument back if you prefer.

Without the sudo, here is the error when running `cmake`: https://travis-ci.org/swarrot/swarrot/jobs/300596815#L706 (it's on swarrot but it's the same install script)

pinepain commented:
@odolbeau that's ok, build setting for CI are not the most obvious part.

Since you are there, could I ask you to alter provision script too - https://github.com/pdezwart/php-amqp/blob/b1e12ee93c3f9693c177ee58444582205d63d581/provision/provision.sh#L81-L84

odolbeau commented:
@pinepain I already updated the provision.sh script

odolbeau commented on .travis.yml:
```diff
@@ -54,8 +54,8 @@ matrix:
-  - sh provision/install_rabbitmq-c.sh ${LIBRABBITMQ_VERSION}
-  - if [ "${COVERITY_SCAN_BRANCH}" != 1 ]; then phpize && ./configure  --with-librabbitmq-dir=$HOME/rabbitmq-c && make ; fi
+  - sudo ./provision/install_rabbitmq-c.sh
+  - if [ "${COVERITY_SCAN_BRANCH}" != 1 ]; then phpize && ./configure && make ; fi
```
I removed the `--with-librabbitmq-dir` option cause the rabbitmq-c library is now installed globally.

pinepain replied:
I guess `${LIBRABBITMQ_VERSION}` was removed by accident. Or there was any other changes to build system I missed?

odolbeau replied:
`${LIBRABBITMQ_VERSION}` is used to access to an environment variable.

Before: this environment variable was passed as an argument in the installation script
Now: The script directly use the environment variable

As said, if you prefer I can update the script to use an argument instead of the environment variable directly.

pinepain replied:
The idea to install rabbitmq-c locally is to avoid using `sudo` in CI so that we could utilize container-based travis build env, [which is faster](https://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/). I guess we already have `sudo: false` in `.travis-ci.yml` so unless there is hard requirement to install globally, I would avoid sudo and go with local install.

pinepain replied:
@odolbeau oh, I'm sorry, missed that. I'd suggest to pass version explicitly. The rationale for that is that this script is copied time to time by others for their CI and for internal  purposes, so I'd prefer to make their life easier.

If you decide to bring argument passing back, could you also add this check to install script

```
if [ "$#" -ne 1 ]; then
  echo "Usage: $0 LIBRABBITMQ_VERSION" >&2
  exit 1
fi
```

odolbeau commented:
OK, travis is still running but it looks like tests are green!
Sorry for all updates I did during your review.
This PR looks good to me right now, feel free to comment it. :)

odolbeau commented:
I added the argument back for the installation script.

Regarding the usage of `sudo`, it looks like the build time is similar to previous build ( Total time 3 hrs 7 min 34 sec ) which made me think it's still used container-based travis build env.

If you want to get rid of `sudo` & global install anyway, I'll be glad to have your help regarding the usage of `clang` to achieve this.

pinepain commented:
@odolbeau thanks a lot, looks decent! Let's merge!

lstrojny commented on provision/provision.sh:
```diff
@@ -69,7 +69,7 @@ sudo service nginx restart
 
 # Install and configure RabbitMQ
-wget -qO - http://www.rabbitmq.com/rabbitmq-signing-key-public.asc | sudo apt-key add -
+wget -qO - http://www.rabbitmq.com/rabbitmq-release-signing-key.asc | sudo apt-key add -
```
Can we switch to https here as well?

odolbeau replied:
Done.

lstrojny commented:
Thank you!
